src/test/java/com/meta/chatbridge/llm/HuggingFaceLlamaPluginTest.java

```diff
@@ -0,0 +1,230 @@
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class HuggingFaceLlamaPluginTest {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    public static final ArrayNode SAMPLE_RESPONSE = MAPPER.createArrayNode();
+    private static final String PATH = "/";
+    private static final String TEST_MESSAGE = "this is a test message";
+    private static final String TEST_ENDPOINT = "test_endpoint";
+    private static final MessageStack<FBMessage> STACK =
+            MessageStack.of(
+                    MessageFactory.instance(FBMessage.class)
+                            .newMessage(
+                                    Instant.now(),
+                                    "test message",
+                                    Identifier.random(),
+                                    Identifier.random(),
+                                    Identifier.random(),
+                                    Role.USER));
+
+    static {
+        ((ArrayNode) SAMPLE_RESPONSE)
+                .addObject()
+                .put("generated_text", TEST_MESSAGE);
+    }
+
+    private BlockingQueue<OutboundRequest> HuggingFaceLlamaRequests;
+    private Javalin app;
+    private URI endpoint;
+    private ObjectNode minimalConfig;
+
+    static Stream<HuggingFaceConfigTest.ConfigItem> modelOptions() {
+        Set<String> non_model_options = Set.of("name", "type", "api_key", "max_input_tokens");
+        return HuggingFaceConfigTest.CONFIG_ITEMS.stream().filter(c -> !non_model_options.contains(c.key()));
+    }
+
+    @BeforeEach
+    void setUp() throws UnknownHostException, URISyntaxException {
+        HuggingFaceLlamaRequests = new LinkedBlockingDeque<>();
+        app = Javalin.create();
+        app.before(
+                PATH,
+                ctx ->
+                        HuggingFaceLlamaRequests.add(
+                                new OutboundRequest(ctx.body(), ctx.headerMap(), ctx.queryParamMap())));
+        app.post(PATH, ctx -> ctx.result(MAPPER.writeValueAsString(SAMPLE_RESPONSE)));
+        app.start(0);
+        endpoint =
+                URIBuilder.localhost().setScheme("http").appendPath(PATH).setPort(app.port()).build();
+    }
+
+    @Test
+    void sampleValid() throws IOException, InterruptedException {
+        String apiKey = UUID.randomUUID().toString();
+        HuggingFaceConfig config = HuggingFaceConfig.builder(apiKey).endpoint(TEST_ENDPOINT).tokenLimit(100).build();
+        HuggingFaceLlamaPlugin<FBMessage> plugin = new HuggingFaceLlamaPlugin<FBMessage>(config).endpoint(endpoint);
+        FBMessage message = plugin.handle(STACK);
+        assertThat(message.message()).isEqualTo(TEST_MESSAGE);
+        assertThat(message.role()).isSameAs(Role.ASSISTANT);
+        assertThatCode(() -> STACK.with(message)).doesNotThrowAnyException();
+        @Nullable OutboundRequest or = HuggingFaceLlamaRequests.poll(500, TimeUnit.MILLISECONDS);
```

Comment:
lets add a snapshot test here for the payload, the payload construction is a little weird so it would be good to ensure it's correct.